src/player.cpp

```diff
@@ -512,17 +514,17 @@ void player::action_taken()
 void player::update_morale()
 {
     // Decay existing morale entries.
-    for( size_t i = 0; i < morale.size(); i++ ) {
+    for( size_t i = morale.size() - 1; i > 0; i-- ) {
```

Comment:
Damn it! You're right! I've messed this condition. It was one of the last quick changes I made before the commit and I should have done proper testing.